caddyhttp/httpserver/replacer.go

```diff
@@ -243,6 +243,11 @@ func (r *replacer) getSubstitution(key string) string {
 	case "{path_escaped}":
 		u, _ := r.request.Context().Value(OriginalURLCtxKey).(url.URL)
 		return url.QueryEscape(u.Path)
+	case "{request_id}":
+		reqid, _ := r.request.Context().Value(RequestIDCtxKey).(string)
+		if reqid != "" {
+			return reqid
```

Comment:
You can elide the `if` check for empty string, because if it's not a value that is set, the string will be empty anyway.